### PR TITLE
fix(web): safely parse trace timestamp query params (#15256)

### LINKED ISSUE #15256: bug: Malformed timestamp query param crashes trace/peek views
## Problem

Trace detail and table-peek views parse the `timestamp` URL query param with bare `new Date(decodeURIComponent(...))`. A malformed percent-encoded value (for example `?timestamp=%`) throws `URIError` and can crash the React tree. Invalid date strings become `Invalid Date` and are passed through inconsistently.

## Motivation

Self-hosted and shared links sometimes carry a corrupted or hand-edited `timestamp` query param. The page should degrade to the existing no-timestamp lookup path instead of throwing.

## Background

Eval preview navigation already guards this in a private `parseTimestamp` helper (`web/src/features/evals/hooks/useEvalPreviewNavigation.ts`): array check, try/catch around `decodeURIComponent`, and `Number.isNaN(date.getTime())`. Peek views and the standalone trace page still use the unsafe pattern. PR #12796 added decoding for peek observation detail but did not add validation.

## Current behavior

- Malformed encoding (`%`) → `URIError` (uncaught)
- Non-date string → `Invalid Date` passed to data hooks
- Array query values are not handled consistently across call sites

## Expected behavior

Parse failures return `undefined` so callers keep the existing no-timestamp code path (correct results; slightly less partition pruning).

## Proposed solution

Extract a shared `parseTraceTimestampFromQuery()` utility matching the eval-preview helper, and use it in:

- `peek-trace-detail.tsx`
- `peek-observation-detail.tsx`
- `peek-experiment-item-detail.tsx`
- `pages/project/[projectId]/traces/[traceId].tsx`
- optionally reuse from `useEvalPreviewNavigation.ts`

Add client unit tests for valid, encoded, malformed, invalid, array, and empty inputs.

## Alternatives considered

- Leave call sites as-is and rely on well-formed links only — still crashes on bad URLs
- Show an error toast on parse failure — heavier UX for a recoverable query param

## Scope

- Client-side query parsing only
- No API, schema, or ClickHouse query changes

## Acceptance criteria

- [ ] Shared parser returns `Date` for valid / URL-encoded ISO timestamps
- [ ] Malformed encoding and invalid dates return `undefined` without throwing
- [ ] Array / empty query values return `undefined`
- [ ] Peek + standalone trace pages use the shared parser
- [ ] Client tests cover the cases above

## Backward compatibility

Compatible. Failure falls back to the same path as "no timestamp in URL".

## Risks

Low. Defensive input sanitization only.

### PR DESCRIPTION
**Pull request by Harsh23Kashyap**
## What does this PR do?

Fixes langfuse/langfuse#15256

Trace/peek views parsed `router.query.timestamp` with bare `new Date(decodeURIComponent(...))`. Malformed percent-encoding (e.g. `?timestamp=%`) throws `URIError` and can crash the page. Invalid dates became `Invalid Date` and were passed through.

This adds `parseTraceTimestampFromQuery()` (same guards as the existing eval-preview helper) and wires peek + standalone trace pages to use it. Parse failures return `undefined` so callers keep the no-timestamp lookup path.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Mandatory Tasks

- [X] Make sure you have self-reviewed the code.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have checked if my changes generate no new warnings
- [X] I have added tests that prove my fix is effective
- [X] I have checked if new and existing unit tests pass locally with my changes

## Testing

```bash
pnpm --filter web run test-client parseTraceTimestampFromQuery.clienttest.ts
# Test Files 1 passed (1), Tests 6 passed (6)
```

<details><summary>Greptile Summary</summary>

This PR safely parses trace timestamp query parameters across trace views. The main changes are:

* Adds a shared parser that rejects malformed encoding and invalid dates.
* Uses the parser in trace, peek, and evaluation-preview navigation paths.
* Adds focused tests for valid, encoded, malformed, repeated, empty, and missing values.

</details>

<details><summary>Confidence Score: 5/5</summary>

This looks safe to merge.

* No blocking issues found in the changed code.

</details>

Reviews (1): Last reviewed commit: ["chore: merge upstream/main"](<https://github.com/langfuse/langfuse/commit/070c3c2c9f2e808d55170eef03e352a8edecea6e>) | [Re-trigger Greptile](<https://app.greptile.com/api/retrigger?id=45944341>)

**Context used:**

* Rule used - Move imports to the top of the module instead of p... ([source](<https://app.greptile.com/personal-org-4986/-/custom-context?memory=c960fc07-9928-409f-a18b-a780cbdded12>))

**Learned From**<br>[langfuse/langfuse-python#1387](<https://github.com/langfuse/langfuse-python/pull/1387>)

### CODE REVIEW

**Comment by nkabardin** [comment]
@claude review

File: web/src/features/evals/hooks/useEvalPreviewNavigation.ts [hunk 1]
```diff
@@ -90,8 +79,8 @@ export function getEvalPreviewPointerFromUrlQuery(
     typeof query.observationId === "string" ? query.observationId : undefined;
   const timestamp =
     target === EvalTargetObject.TRACE
-      ? parseTimestamp(query.timestamp)
-      : (parseTimestamp(query.startTime) ?? parseTimestamp(query.timestamp));
+      ? parseTraceTimestampFromQuery(query.timestamp)
+      : (parseTraceTimestampFromQuery(query.startTime) ?? parseTraceTimestampFromQuery(query.timestamp));
 
   if (target === EvalTargetObject.TRACE) {
```
[the remark this answers is not in the document]
Comment (nkabardin) [reply web/src/features/evals/hooks/useEvalPreviewNavigation.ts:86]: @Harsh23Kashyap would you reformat please?

**Comment by Harsh23Kashyap** [comment]
Pushed in `d1b59a38a`. Prettier wanted the `??` chain on multiple lines, applied the canonical form, and `pnpm exec prettier --check` now passes on the file.

File: web/src/features/evals/hooks/useEvalPreviewNavigation.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Harsh23Kashyap) [reply web/src/features/evals/hooks/useEvalPreviewNavigation.ts:86]: Pushed in `d1b59a38a`. Prettier wanted the `??` chain on multiple lines, applied the canonical form, and `pnpm exec prettier --check` now passes on the file.
